native/cocos/audio/android/AudioEngine-inl.cpp

```diff
@@ -396,3 +397,90 @@ void AudioEngineImpl::onResume() {
         _audioPlayerProvider->resume();
     }
 }
+
+uint32_t AudioEngineImpl::getSampleRate(const char* url) {
+    return outputSampleRate;
+}
+
+float reduceFactor(AudioDataType type) {
+    float ret = 1.0F;
+    switch (type) {
+        case cc::AudioDataType::SIGNED_16:
+            ret = 1.0F / static_cast<float>(SHRT_MAX);
+            break;
+        case cc::AudioDataType::SIGNED_32:
+            ret = 1.0F / static_cast<float>INT32_MAX;
+            break;
+        case cc::AudioDataType::FLOAT_32:
+            //TODO: check the reduceFactor of float 32 to reset [-1,1], most of time is 1.0f
+            ret = 1.0F;
+        default:
+            break;
+    }
+    return ret;
+}
+using AudioDataInterpreter = std::function<float (char *)>;
+AudioDataInterpreter getFormatedData(AudioDataType type) {
```

Comment:
c functions need to be at the beginning of cpp file and  in anonymous namespace.

Reply:
1